pytorch_lightning/utilities/enums.py

```diff
@@ -31,6 +31,11 @@ def __eq__(self, other: Union[str, Enum]) -> bool:
         other = other.value if isinstance(other, Enum) else str(other)
         return self.value.lower() == other.lower()
 
+    def __hash__(self):
+        # re-enable hashtable so it can be used as dict key or form a set
+        # example: set(LightningEnum)
+        return hash(self._name_)
```

Comment:
```suggestion
        # re-enable hashtable so it can be used as a dict key or in a set
        # example: set(LightningEnum)
        return hash(self.name)
```